docs/course_materials/compound_data_types.html

```diff
@@ -189,32 +244,171 @@ <h1><b>
+                        print(b) # 2
+                        print(c) # 3
                     </code></pre>
                 </section>
                 <section data-background-size="cover" data-background-image="media/exr_background.jpeg" data-markdown="exercises/04_Compound_data_types/class_exercise_2.md" data-separator="^\n\n\n" data-separator-vertical="^\n\n">
                 </section>
             </section>
+            <!-- Slicing -->
+            <section>
+                <section>
+                    <h1>
+                        <b><span>Slicing</span></b>
+                    </h1>
+                </section>
+                <section>
+                    <h3>
+                        <b>Overview</b>
+                    </h3>
+                    <h4>
+                        Slicing is a feature that extracts sub-sets out of lists, tuples and arrays
+                    </h4>
+                </section>
+                <section>
+                    <h3>
+                        <b>Syntax</b>
+                    </h3>
+                    <pre><code data-trim data-noescape class="hljs lang-python">
+                            my_list[start:end:step]
+                    </code></pre>
+                    <ul>
+                        <li>start - The index from which the sub-list starts (inclusive)</li>
+                        <li>end - The index at which the sub-list ends (exclusive)</li>
+                        <li>step - How many elements to skip</li>
+                    </ul>
+                </section>
+                <section>
+                    <h3>
+                        <b>Examples</b>
+                    </h3>
+                    <pre><code data-trim data-noescape class="hljs lang-python">
+                        my_list = ['a', 'b', 'c', 'd', 'e']
+                        my_list[1:3] # ['b', 'c']
+                        my_list[1:] # ['b', 'c', 'd', 'e']
+                        my_list[:3] # ['a', 'b', 'c']
+                        my_list[:] # ['a', 'b', 'c', 'd', 'e']
+                        my_list[::2] # ['a', 'c', 'e']
+                    </code></pre>
+                </section>
+                <section>
+                    <h4>Important:</h4>
+                    <ul>
+                        <li>The <b>start</b> element is inclusive, which means that it will be included in the sub-list</li>
+                        <li>The <b>end</b> element is exclusive, which means that it won't be included in the sub-list</li>
+                        <li>The <b>step</b> element equals to 1 by default and doesn't need to be written always</li>
+                    </ul>
+                    <h4>Example:</h4>
+                    <pre><code data-trim data-noescape class="hljs lang-python">
+                        my_list[1:3] # ['b', 'c']
+                        #      [start    ]end
+                        #   0 ,  1 ,  2 ,  3 ,  4
+                        # ['a', 'b', 'c', 'd', 'e']
```

Comment:
The example code seems a bit messed up (a lot of spaces after the start).
Maybe it's better to edit and image and add it here

Reply:
The spaces are intentionally there to indicate which indices are included and which are not.
What do you think?

Reply:
Otherwise it's hard to illustrate that without a real picture

Reply:
you can do something as I did:
https://github.com/RedHat-Israel/ROSE/blob/master/docs/course_materials/media/functions.png